dbldatagen/column_generation_spec.py

```diff
@@ -89,7 +87,9 @@ class ColumnGenerationSpec(object):
 
     def __init__(self, name, colType=None, minValue=0, maxValue=None, step=1, prefix='', random=False,
                  distribution=None, baseColumn=None, randomSeed=None, randomSeedMethod=None,
-                 implicit=False, omit=False, nullable=True, debug=False, verbose=False, **kwargs):
+                 implicit=False, omit=False, nullable=True, debug=False, verbose=False,
```

Comment:
This is hard to maintain/develop. Probably these properties relate to various behavioural objects. 
You can separate them using pydantic or dataclass, for instance:

```python

from pydantic import BaseModel

class PropertySetNumberOne(BaseModel):
    col_type: Optional[str] = None
    min_value: Optional[int] = 1
 
 class PropertySetNumberOne:
    implicit: bool = False
    omit: bool = False
    
 class MyNiceClass: 
    ....
    # add here implementation for .with methods
 
# example class invocation

obj_instance = (
    MyNiceClass
        .withPropertySet(PropertySet1())
        .withAnotherPropertySet(PropertySet2())
)
```

Also the variable naming is inconsistent - input parameters are `camelCase`, but default is snake_case (e.g. `DEFAULT_SEED_COLUMN`.

Reply:
Agreed but not part of code changes for this release

Actually code does use options object but as this was used in some high value deals even in prerelease version, did not want to change API signatures

Reply:
Also - want to only use frameworks used by DB Runtime releases